### PR TITLE
Addon Automation Part 1 [buildNum:2024.02.21.20]

### PR DESCRIPTION
**Pull request by DanRod1999**
works alongside: https://github.com/iTwin/imodel-native-internal/pull/379

### CODE REVIEW

File: build/begin_addon_release.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            echo "##vso[task.setvariable variable=ITWINJS_BRANCH;isoutput=true]$iTwinjsBranchName"
+            echo "##vso[task.setvariable variable=NATIVE_BUILD_NUM;isoutput=true]imodel-native-$nativeBuildNum"
+
+            noCi="[no ci]"
+            if [[ $(Build.SourceVersionMessage) != *"$noCi"* ]]; then
+              echo "##vso[task.setvariable variable=SKIP_ADDON;isoutput=true]true"
+            fi
+        env:
+          GITHUB_TOKEN: $(GH_TOKEN)
+        displayName: get build id from commit
```
Comment (nick4598) [inline build/begin_addon_release.yml:55]: Nitpick but maybe we should switch this to build num instead?

File: build/begin_addon_release.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+              iTwinjsBranchName='master'
+            else
+              iTwinjsBranchName=$(gh pr view $itwinjsPR --json headRefName  --repo iTwin/itwinjs-core -q '.headRefName')
+            fi
+
+            echo "##vso[task.setvariable variable=ITWINJS_BRANCH;isoutput=true]$iTwinjsBranchName"
+            echo "##vso[task.setvariable variable=NATIVE_BUILD_NUM;isoutput=true]imodel-native-$nativeBuildNum"
+
+            noCi="[no ci]"
+            if [[ $(Build.SourceVersionMessage) != *"$noCi"* ]]; then
```
[the remark this answers is not in the document]
Comment (nick4598) [reply build/begin_addon_release.yml:50]: Native PR is Linked and current PR is imodel-native
            if [[ "${{ parameters.otherRepo }}" == "imodel-native-internal" ]];then
              echo 'Skipping build in imodel-native. Build will proceed in imodel-native-internal'

              echo 'Updating PR title to skip Addon Release'
              noCi="[no ci]"
              if [[ $title != *"$noCi"* ]]; then
                gh pr edit $branch --title "$title [no ci]"
              fi

              Nvm I found it

[the remark this answers is not in the document]
Comment (nick4598) [reply build/begin_addon_release.yml:50]: Maybe worth mentioning it gets added by 'pipeline.yml' in imodel-native-internal for x reason

File: build/azure-pipelines.yml [hunk 3]
```diff
@@ -7,7 +8,7 @@ trigger:
       - release/*
 pr:
   autoCancel: true
-  drafts: false
+  drafts: true #FIXME: revert to true afters
```
Comment (aruniverse) [inline build/azure-pipelines.yml:11]: ?

**Review by nick4598 — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Pull request merged**